pyodide-build/pyodide_build/mkpkg.py

```diff
@@ -186,14 +217,26 @@ def update_package(package: str, update_patched: bool = True):
         else:
             abort(f"Pyodide applies patches to {package}. Skipping update.")
 
-    sdist_metadata = _extract_sdist(pypi_metadata)
+    dist_metadata = None
+    if not wheel and not sdist:
+        wheel = True
+        sdist = True
+        if not yaml_content["source"]["url"].endswith("whl"):
+            # If it is currently an sdist, prefer to update to an sdist
+            dist_metadata = _find_sdist(pypi_metadata)
+
+    if not dist_metadata:
+        dist_metadata = _find_dist(pypi_metadata, wheel, sdist)
 
-    yaml_content["source"]["url"] = sdist_metadata["url"]
+    assert dist_metadata
```

Comment:
maybe let's rather raise an ValueError with an explicit message?

Reply:
`_find_dist` never returns `None` instead it raises a `MkpkgFailedException`, so it should be impossible for this assertion to fail. It is just here to convince mypy that `dist_metadata` is not `None`.

Reply:
Okay, so the better solution here is to fix the return type of `_find_dist`.